src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -565,12 +586,22 @@ where
                     source.source_id()
                 )
             } else {
-                bail!(
-                    "could not find `{}` in {} with version `{}`",
-                    dep.package_name(),
-                    source.source_id(),
-                    dep.version_req(),
-                )
+                if was_git_url_miscategorised_as_a_registry_dep(&dep) {
+                    bail!(
+                        "could not find `{}` in {} with version `{}`. Try adding `--git {}`",
```

Comment:
While I don't think I've seen cargo strictly follow rustc's error conventions, I think it'd be helpful for us to consider them with our error messages.  See https://rustc-dev-guide.rust-lang.org/diagnostics.html#suggestion-style-guide.

While this isn't quite a question, it still falls under the "be more explicit" part of questions where it says

> Sometimes, it's unclear why a particular suggestion is being made

Maybe the "further instruction" section can also provide ideas on how to work this.

> The message may contain further instruction such as "to do xyz, use" or "to do xyz, use abc".

Reply:
Thanks for pointing this out. I will rewrite the error message to follow rustc conventions for the suggestion along the lines of 

> To install a package from a git repository, use `--git {}`

I think the original first part of the error message is a) good enough b) familiar to users by now, so will only rewrite the suggestion